exp-1-merit/kafka-shared/modules/consumer/variables.tf

```diff
@@ -0,0 +1,100 @@
+# ACL
+variable "topic_acl_resource_name" {
+  type        = string
+  description = "The name of the Kafka resource for which the ACL is being created."
+}
+
+variable "group_acl_resource_name" {
+  type        = string
+  description = "The name of the Kafka resource for which the ACL is being created."
+}
+
+variable "topic_acl_resource_type" {
+  type        = string
+  description = "The type of the Kafka resource for which the ACL is being created. Valid values: 'Unknown', 'Any', 'Topic', 'Group', 'Cluster', or 'TransactionalID'."
+  default     = "Topic"
+
+  validation {
+    condition     = contains(["Unknown", "Any", "Topic", "Group", "Cluster", "TransactionalID"], var.topic_acl_resource_type)
+    error_message = "Invalid resource_type. Must be one of: 'Unknown', 'Any', 'Topic', 'Group', 'Cluster', or 'TransactionalID'."
+  }
+}
+
+variable "group_acl_resource_type" {
+  type        = string
+  description = "The type of the Kafka resource for which the ACL is being created. Valid values: 'Unknown', 'Any', 'Topic', 'Group', 'Cluster', or 'TransactionalID'."
+  default     = "Group"
+
+  validation {
+    condition     = contains(["Unknown", "Any", "Topic", "Group", "Cluster", "TransactionalID"], var.group_acl_resource_type)
+    error_message = "Invalid resource_type. Must be one of: 'Unknown', 'Any', 'Topic', 'Group', 'Cluster', or 'TransactionalID'."
+  }
+}
+
+variable "indexer_user" {
+  type        = string
+  description = "The principal or user/group for which the ACL is being created."
+}
+
+variable "acl_host" {
+  type        = string
+  description = "The host from which the principal is allowed to perform operations. Set to '*' to allow any host."
+  default     = "*"
+}
+
+variable "acl_operation" {
+  type        = string
+  description = "The operation that is allowed on the resource. Valid values: 'Unknown', 'Any', 'All', 'Read', 'Write', 'Create', 'Delete', 'Alter', 'Describe', 'ClusterAction', 'DescribeConfigs', 'AlterConfigs', or 'IdempotentWrite'."
+  default     = "Read"
+
+  validation {
+    condition     = contains(["Unknown", "Any", "All", "Read", "Write", "Create", "Delete", "Alter", "Describe", "ClusterAction", "DescribeConfigs", "AlterConfigs", "IdempotentWrite"], var.acl_operation)
+    error_message = "Invalid acl_operation. Must be one of: 'Unknown', 'Any', 'All', 'Read', 'Write', 'Create', 'Delete', 'Alter', 'Describe', 'ClusterAction', 'DescribeConfigs', 'AlterConfigs', or 'IdempotentWrite'."
+  }
+}
+
+variable "acl_permission_type" {
+  type        = string
+  description = "The permission type for the ACL. Valid values: 'Unknown', 'Any', 'Allow', or 'Deny'."
+  default     = "Allow"
+
+  validation {
+    condition     = contains(["Unknown", "Any", "Allow", "Deny"], var.acl_permission_type)
+    error_message = "Invalid acl_permission_type. Must be one of: 'Unknown', 'Any', 'Allow', or 'Deny'."
+  }
+}
+
+# QUOTA
+variable "entity_type" {
```

Comment:
I don't think we need a var for this, as it should always be `user` with our setup.